src/simulation/simulation.jl

```diff
@@ -434,20 +423,40 @@ function _apply_warm_start!(model::DecisionModel)
 end
 
 """ Required update problem function call"""
-function _update_problem!(model::DecisionModel, sim::Simulation)
+function _update_model!(model::DecisionModel, sim::Simulation)
     _update_parameters(model, sim)
     _update_initial_conditions!(model, sim)
     return
 end
 
+function _update_simulation_state!(sim::Simulation)
+    store = get_simulation_store(sim)
+    simulation_time = get_current_time(sim)
+    state = get_simulation_state(sim)
+    for model_name in list_models(store), type in [:variables, :aux_variables]
+        model_params = get_model_params(store, model_name)
+        for key in list_fields(store, model_name, type)
+            res = read_result(
+                JuMP.Containers.DenseAxisArray,
+                store,
+                model_name,
+                key,
+                simulation_time,
+            )
+            @show model_name key
```

Comment:
Stale? And next line is commented, but maybe that is intentional for current progress.